### DISCUSSION TITLE
Using tensorboard instead of wandb

### DISCUSSION THREAD

**Discussion by pratikkumar018** [OpenRLHF/OpenRLHF#234] [Q&A] [answered]
Hi Team, 
I wanted to use tensorboard instead of wandb. Is there any flag or way i can quickly make this switch. I am running on offline servers.

**Comment by wuxibin89** [comment]
@pratikkumar018 Hi, we don't support tensorboard for now. Can you help to contribute? it's super easy to replace wandb with tensorboard:
https://github.com/OpenLLMAI/OpenRLHF/blob/main/openrlhf/trainer/ppo_trainer.py#L362-L370

**Reply by pratikkumar018** [reply]
Sure will check. Thanks

**Reply by pratikkumar018** [reply]
Hi Here is the code I have written for tensorboard


https://github.com/OpenLLMAI/OpenRLHF/blob/main/openrlhf/trainer/ppo_trainer.py#L147
```       
if self.strategy.args.use_tensorboard and self.strategy.is_rank_0():
              from torch.utils.tensorboard import SummaryWriter
              current_time = datetime.now().strftime("%Y%m%d-%H%M%S")
              self._writer=SummaryWriter('{}/{}'.format(log_dir, current_time))

```
https://github.com/OpenLLMAI/OpenRLHF/blob/4e15591a5abd19a14e4a72415603bce76c3e1567/openrlhf/trainer/ppo_trainer.py#L371
```            if self._writer is not None and self.strategy.is_rank_0():

                logs = {
                    "train/%s" % k: v
                    for k, v in {
                        **logs_dict,
                    }.items()
                }
                                
                for k, v in logs.items():
                    print(k,v)
                    print(global_step)
                    self._writer.add_scalar(k, v, global_step=global_step)
                self._writer.add_scalar('global_step', global_step, global_step=global_step)
```


https://github.com/OpenLLMAI/OpenRLHF/blob/4e15591a5abd19a14e4a72415603bce76c3e1567/openrlhf/trainer/ppo_trainer.py#L195
```        
if self.strategy.args.use_tensorboard and self.strategy.is_rank_0():
            self._writer.close()```

**Reply by pratikkumar018** [reply]
@wuxibin89 can you please help me with this for ppo_ray as well. I tried finding couldnt locate the file where it needs to be done. I need to use this for bigger model . Or  since it is already inheriting PPO trainer changing it there would be enough?

**Reply by wuxibin89** [reply, marked as answer]
ppo_ray inherits PPO trainer, so it should be enough.

**Reply by pratikkumar018** [reply]
Thanks.

**Reply by wuxibin89** [reply]
@pratikkumar018 Hi, can you contribute your tensorboard support with a PR? We're very appreciate it.
